### DISCUSSION TITLE
Grasscutter | v1.2.0

### DISCUSSION THREAD

**Discussion by KingRainbow44** [Melledy/Grasscutter#1234] [Announcements]
ooh, another Grasscutter update... _how interesting..._
Development has slowed down since 1.0.0, but there are still features nonetheless.
Thanks for the support relating to the Grasscutter projects!

## Changes (short)
- Update READMEs
- Add `lombok` as a dependency
- Add a `BuildConfig.java` file into the JAR containing the artifact details
- Create a generic Grasscutter installer for UNIX systems
- Update `kcp-netty`
- Add a JSON schema for Grasscutter plugins
- Update to anime game `2.7`
- Update `.proto` files to support anime game `2.7`
- Overhaul the configuration system
- Replace the dispatch server with the HTTP server
- Overhaul the authentication system
- Add a requirement for command targets
- Add method for sending a default command's translated messages
- Ensure the player for an account is kicked before deleting the account
- Fix & enhance functionality of commands
- Add a command for changing a specific user's language
- Implement stamina
- Implement quests
- Implement a spawning limit
- Add a command for modifying a player's team(s)
- Implement Spiral Abyss
- Add a command to skip Spiral Abyss floors
- Implement abilities(?)
- Implement crafting
- Implement domains
- Implement gadgets (item gadgets or entity gadgets)
- Implement expeditions
- Implement the character friendship system
- Overhaul the gacha system
- Implement deforestation

## What's Changed
(these are PRs from `1.1.0` -> `1.2.0`)
* another translation patches because i fucked it up by @Kimi898246 in https://github.com/Grasscutters/Grasscutter/pull/652
* More reliable stamina calculation by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/625
* Monsters spawned turn by turn by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/630
* Polish language by @zakhil-dev in https://github.com/Grasscutters/Grasscutter/pull/655
* Include all `grasscutter.jar` files in Actions Build by @KingRainbow44 in https://github.com/Grasscutters/Grasscutter/pull/658
* Add a plugin schema by @KingRainbow44 in https://github.com/Grasscutters/Grasscutter/pull/656
* Fix: null reference on stamina manager timer when paused player disconnects by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/663
* fix issues 646 by @Mateoust in https://github.com/Grasscutters/Grasscutter/pull/665
* Fix typo && update zh-CN.json by @Thomitics in https://github.com/Grasscutters/Grasscutter/pull/660
* Fix uploading by @Thomitics in https://github.com/Grasscutters/Grasscutter/pull/670
* Add command description multilingual by @577fkj in https://github.com/Grasscutters/Grasscutter/pull/659
* Override server logging level with environment variable in https://github.com/Grasscutters/Grasscutter/pull/653
* StaminaManager update by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/680
* Complete the description in the help command by @chrisblue in https://github.com/Grasscutters/Grasscutter/pull/682
* Ability invoke handler by @Melledy in https://github.com/Grasscutters/Grasscutter/pull/690
* Persist Tower Data && Set The Tower Schedule  by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/676
* feature(task): Implement pause, resume and cancel by @Yazawazi in https://github.com/Grasscutters/Grasscutter/pull/691
* Change LICENSE to GNU Affero by @memetrollsXD in https://github.com/Grasscutters/Grasscutter/pull/684
* Update com.google.code.gson 2.8.8 -> 2.9.0 by @memetrollsXD in https://github.com/Grasscutters/Grasscutter/pull/702
* Fallback the Fallback by @KingRainbow44 in https://github.com/Grasscutters/Grasscutter/pull/703
* Added to ability to specify main and substats for /giveart via names instead of IDs by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/704
* Change the license type in zh-CN.json. by @Thomitics in https://github.com/Grasscutters/Grasscutter/pull/711
* Target permissions by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/575
* Support Team Toggle in Tower & Refactor MonsterTide by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/718
* [FIX] Stamina: only handle motion notify for current entity. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/721
* [Enhance] Exclude macOS junk in case anyone develops on mac. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/722
* Fix #719  by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/723
* [Fix] disabling stamina in config. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/728
* Fix /talent saving to level instead of id by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/730
* fix issues #648 #677 by @Mateoust in https://github.com/Grasscutters/Grasscutter/pull/731
* Fixed MissingFormatArgumentException (#727) by @jie65535 in https://github.com/Grasscutters/Grasscutter/pull/735
* fix the Monster spawn between stage challenges by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/746
* Change `account delete` to delete all data elements referenced by the deleted account. by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/747
* fix error KillCharacter description in the help command by @CCasusensa in https://github.com/Grasscutters/Grasscutter/pull/741
* Disable falling damage for godmode by @hatsune-miku in https://github.com/Grasscutters/Grasscutter/pull/750
* Update build.gradle by @RealHeart in https://github.com/Grasscutters/Grasscutter/pull/764
* Fix the decision statement of adding map marker. by @341101 in https://github.com/Grasscutters/Grasscutter/pull/763
* Fix the following issues: by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/726
* [Enhance] No more fallen to death in god mode by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/768
* [Fix] Stamina will be set to full if Stamina is disabled. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/769
* [Enhance] Temporarily set statue auto use to 1 and 100%. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/770
* Fix delete account can not delete all related data by @EndEdge in https://github.com/Grasscutters/Grasscutter/pull/772
* Update zh-CN.json and fix typo by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/774
* fix killCharacter command description typo by @CCasusensa in https://github.com/Grasscutters/Grasscutter/pull/777
* [L10N] Add UTF8 charset in dispatch server HTML template. by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/779
* [Feature] Talent moving stamina cost by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/775
* add /language command by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/780
* fixed pr780, uploaded LanguageCommand.java by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/782
* Improve text by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/784
* Fix language switching prompt and save by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/788
* Improve logging pattern by including caller class name in https://github.com/Grasscutters/Grasscutter/pull/688
* [Feature] Stamina: claymore charged attacks by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/795
* Implement a better config structure by @KingRainbow44 in https://github.com/Grasscutters/Grasscutter/pull/802
* Update Issue template to ask for version by @mingjun97 in https://github.com/Grasscutters/Grasscutter/pull/799
* [Fix] incorrect ascension level in givechar command by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/810
* Fix new config infra bugs by @mingjun97 in https://github.com/Grasscutters/Grasscutter/pull/806
* fixed language not found prompt by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/807
* Add Server status API for three-party game launcher by @Coooookies in https://github.com/Grasscutters/Grasscutter/pull/811
* Removed invalid code in getLanguageFileDescription,When the language is not discovered, it will use the built-in language fallback mechanism to fall back,At the same time, we also fix the issue that using language in the server does not save the settings of the server side locale by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/813
* Fix InRouting on dispatch server by @lilmayofuksu in https://github.com/Grasscutters/Grasscutter/pull/826
* It will use the english as default rather than tell you the value is … by @Secretboy-SMR in https://github.com/Grasscutters/Grasscutter/pull/823
* [Plugin API Support] Add verifyUser to AuthenticationHandler by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/819
* Improve text by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/814
* Introduce `-version` option to display version by @mingjun97 in https://github.com/Grasscutters/Grasscutter/pull/797
* Gacha: Banner Details Page by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/829
* [Feature/Fix] Update stamina manager by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/825
* Inventory.payItems() by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/824
* feature(widget): simple implementation of widget by @Yazawazi in https://github.com/Grasscutters/Grasscutter/pull/837
* Clean up: map mark related code by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/841
* Slowly recover HP near statue by @gentlespoon in https://github.com/Grasscutters/Grasscutter/pull/842
* Allow to override final JAR filename using project property in https://github.com/Grasscutters/Grasscutter/pull/852
* Build region_list and cur_region dynamically in https://github.com/Grasscutters/Grasscutter/pull/850
* add docs for tower command by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/844
* Improve text by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/843
* Fix UTF-8 encoding and content in announcements in https://github.com/Grasscutters/Grasscutter/pull/854
* Fix broken time formatting on gacha records in https://github.com/Grasscutters/Grasscutter/pull/855
* fix: LEAK: ByteBuf.release() was not called by @Akka0 in https://github.com/Grasscutters/Grasscutter/pull/857
* [Config-breaking] Gacha revamp by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/639
* Don't silently overwrite config.json if there is an error. by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/863
* Traditional Chinese | Translation Patches by @Kimi898246 in https://github.com/Grasscutters/Grasscutter/pull/864
* Implement quests by @Melledy in https://github.com/Grasscutters/Grasscutter/pull/816
* feature(serenitea pot): Implementation of the entry function by @Yazawazi in https://github.com/Grasscutters/Grasscutter/pull/869
* added support for  codex quests by @ShigemoriHakura in https://github.com/Grasscutters/Grasscutter/pull/870
* Improve text by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/867
* add support for announcement page by @ShigemoriHakura in https://github.com/Grasscutters/Grasscutter/pull/872
* add negative permission check by @exzork in https://github.com/Grasscutters/Grasscutter/pull/874
* Update zh-TW.json by @luluxiaoyu in https://github.com/Grasscutters/Grasscutter/pull/875
* Fix connot execute quest command in console by @zrll12 in https://github.com/Grasscutters/Grasscutter/pull/882
* new command join and remove to force join or remove avatars into your current team by @lsCoding666 in https://github.com/Grasscutters/Grasscutter/pull/549
* Refactor 'dispatch' server/Update plugin authentication API by @KingRainbow44 in https://github.com/Grasscutters/Grasscutter/pull/868
* Add a new command "nostamina" by @tiantian520tt in https://github.com/Grasscutters/Grasscutter/pull/877
* Properly handle static assets for announcements in https://github.com/Grasscutters/Grasscutter/pull/891
* Move gacha files to separate directory and refactor file serving in https://github.com/Grasscutters/Grasscutter/pull/892
* fix permission command by @exzork in https://github.com/Grasscutters/Grasscutter/pull/898
* Fixed grammar in README.md #740 by @liamsegura in https://github.com/Grasscutters/Grasscutter/pull/742
* fix join command and remove command bug and update docs by @lsCoding666 in https://github.com/Grasscutters/Grasscutter/pull/904
* [PLUGIN] Give plugin ability to replace ChatManager by @exzork in https://github.com/Grasscutters/Grasscutter/pull/905
* Joinremove by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/907
* Remove the extra slash while loading resources. by @line-bear in https://github.com/Grasscutters/Grasscutter/pull/893
* Update opcodes and names by @elap5e in https://github.com/Grasscutters/Grasscutter/pull/940
* Fix "/" Route by @6ixfalls in https://github.com/Grasscutters/Grasscutter/pull/928
* add config to enable/disable console by @exzork in https://github.com/Grasscutters/Grasscutter/pull/946
* Optimize the join and remove commands to team by @RDsatan in https://github.com/Grasscutters/Grasscutter/pull/954
* Enhance the default permissions logic by @omg-xtao in https://github.com/Grasscutters/Grasscutter/pull/929
* "Autogenerate" data files with data fallbacks and moved keys folder into jar resources by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/927
* Permanent fix for version.json stuff (fixes 4206 on some clients) by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/962
* Cache gradle files by @DancingSnow0517 in https://github.com/Grasscutters/Grasscutter/pull/885
* Fix "illegal data type conversion to int" and change BANNED_PACKETS to HashSet by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/961
* Fix usage & remove outdated text by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/960
* Do not add default region if there are already regions by @YukariChiba in https://github.com/Grasscutters/Grasscutter/pull/957
* verifyUser now returns the user's Account instead of a boolean by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/964
* Update the Issue template by @lunaticwhat in https://github.com/Grasscutters/Grasscutter/pull/965
* add CommandResponseEvent by @exzork in https://github.com/Grasscutters/Grasscutter/pull/967
* Introduce OAuth in AuthenticationSystem by @YukariChiba in https://github.com/Grasscutters/Grasscutter/pull/956
* General housekeeping and clarification of en-us.json file by @AZthemute in https://github.com/Grasscutters/Grasscutter/pull/968
* Improve text & update zh-CN.json by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/975
* Fix punctuation & update zh-CN.json by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/976
* Fix invalid log archive file format error by @line-bear in https://github.com/Grasscutters/Grasscutter/pull/983
* Add '-web-documentation' to start a simple web server with documentation by @2bllw8 in https://github.com/Grasscutters/Grasscutter/pull/922
* Autogenerate web document files by @omg-xtao in https://github.com/Grasscutters/Grasscutter/pull/988
* Create installer script for Debian/Ubuntu by @TurtleIdiot in https://github.com/Grasscutters/Grasscutter/pull/985
* Remove outdated text & update zh-CN.json by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/995
* Fix connection bugs without removing binding address  by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/993
* Set the maximum number of player on the server through the config.json by @Coooookies in https://github.com/Grasscutters/Grasscutter/pull/1001
* Server Log Events by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/996
* fix: Error when checking files by @zrll12 in https://github.com/Grasscutters/Grasscutter/pull/1013
* Development by @nyakakun in https://github.com/Grasscutters/Grasscutter/pull/1012
* Attempt to fix untranslated login window by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/1015
* Fix dataloader not getting path correctly by @omg-xtao in https://github.com/Grasscutters/Grasscutter/pull/1017
* More Web Static Languages by @4Benj in https://github.com/Grasscutters/Grasscutter/pull/1016
* Traditional Chinese | Translation Patches by @Kimi898246 in https://github.com/Grasscutters/Grasscutter/pull/1025
* Allow commands to target offline players by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/1022
* Update translation keys by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1034
* Implement Some Energy-related Stuff by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1029
* Traditional Chinese | Translation Patch by @CCasusensa in https://github.com/Grasscutters/Grasscutter/pull/1036
* Fix team command by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/1035
* Recommit support for codex(aka archive) animal, weapon, reliquary by @ShigemoriHakura in https://github.com/Grasscutters/Grasscutter/pull/1037
* Add support for installing on arch linux systems to install.sh by @TurtleIdiot in https://github.com/Grasscutters/Grasscutter/pull/1045
* Fix the bug that can't kill command-generated monsters by @ShigemoriHakura in https://github.com/Grasscutters/Grasscutter/pull/1042
* Fix new account can't load codex datas by @ShigemoriHakura in https://github.com/Grasscutters/Grasscutter/pull/1048
* Java not found in start.cmd if you don't installed java 8 by @Apexo01 in https://github.com/Grasscutters/Grasscutter/pull/1049
* Fix usage & update zh-CN.json by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1050
* Implement persist energy by @Logictc in https://github.com/Grasscutters/Grasscutter/pull/1053
* Energy: Particle/Orb Drops from Monsters by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1054
* Skill particle generation by @Logictc in https://github.com/Grasscutters/Grasscutter/pull/1060
* Energy: Fix Amber not Generating Particles by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1061
* fix monster no melee attack and adjust skill particle amount by @Logictc in https://github.com/Grasscutters/Grasscutter/pull/1063
* TeamManager avatar add refactor by @ProxyismGH in https://github.com/Grasscutters/Grasscutter/pull/1064
* Energy: Make sure we only process energy drops for actual monsters. by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1073
* SetStats command feedback string by @Birdulon in https://github.com/Grasscutters/Grasscutter/pull/1079
* Update translation keys by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1080
* Complement usage by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1084
* CoopCommand: Fix usage by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1088
* Hunting and deforestation support. by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1083
* Update Grasscutter to 2.7 by @memetrollsXD in https://github.com/Grasscutters/Grasscutter/pull/1078
* Update 2.7 Banners by @elap5e in https://github.com/Grasscutters/Grasscutter/pull/1121
* fix banner title display by @elap5e in https://github.com/Grasscutters/Grasscutter/pull/1123
* Update default config to match the one in Koko-boya/Grasscutter_Resources by @lhhxxxxx in https://github.com/Grasscutters/Grasscutter/pull/1125
* Usage: Unified specification by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1127
* Fix stuck when talking to npc seller by @line-bear in https://github.com/Grasscutters/Grasscutter/pull/1128
* Language: Update translation by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1133
* Fix deforestation protocol on 2.7 by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1136
* Fix deforestation protocol and add a new field by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1137
* WeatherCommand: Make it clear by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1139
* Fix receiving reward repeatedly by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1140
* Account: Make sure we don't reserve a UID that is already taken by another Player by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1145
* Adding toggle feature to "nostamina" command similar to "godmode" command by @tamilpp25 in https://github.com/Grasscutters/Grasscutter/pull/1152
* Arresting small creature support by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1150
* Fix irreversible equipping by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1169
* Handle unlocking of forging blueprints. by @kyoko12 in https://github.com/Grasscutters/Grasscutter/pull/1165
* Incompletely implement about QuickUseWidgetReq by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1171
* skipping right characters panel for Android by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1175
* Add coordinate support for spawn command (monster only) by @saucebing in https://github.com/Grasscutters/Grasscutter/pull/1177
* Spawn default monsters with different levels by @saucebing in https://github.com/Grasscutters/Grasscutter/pull/1176
* Fix default Shop.json (for Mondstadt General Goods) by @leonier-test in https://github.com/Grasscutters/Grasscutter/pull/1178
* Fix Ruin Guard/Hunter/Grader drops in Drops.json by @leonier-test in https://github.com/Grasscutters/Grasscutter/pull/1179
* Language: Update translation by @Tesutarin in https://github.com/Grasscutters/Grasscutter/pull/1180
* Text fixes by @AZthemute in https://github.com/Grasscutters/Grasscutter/pull/1184
* List game & server version when starting server by @AZthemute in https://github.com/Grasscutters/Grasscutter/pull/1181
* add new command (unlimitenergy):toggle energyusage for each player by @liizfq in https://github.com/Grasscutters/Grasscutter/pull/1186
* Fix unable to save game data occasionally by @zhaodice in https://github.com/Grasscutters/Grasscutter/pull/1194
* Merge `development` into `stable` by @Melledy in https://github.com/Grasscutters/Grasscutter/pull/1122

## New Contributors
* @Mateoust made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/665
* @chrisblue made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/682
* @hatsune-miku made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/750
* @EndEdge made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/772
* @Tesutarin made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/774
* @luluxiaoyu made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/875
* @zrll12 made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/882
* @lsCoding666 made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/549
* @tiantian520tt made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/877
* @liamsegura made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/742
* @elap5e made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/940
* @6ixfalls made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/928
* @RDsatan made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/954
* @YukariChiba made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/957
* @AZthemute made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/968
* @2bllw8 made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/922
* @TurtleIdiot made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/985
* @nyakakun made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1012
* @Apexo01 made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1049
* @Logictc made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1053
* @ProxyismGH made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1064
* @tamilpp25 made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1152
* @saucebing made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1177
* @leonier-test made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1178
* @liizfq made their first contribution in https://github.com/Grasscutters/Grasscutter/pull/1186

**Full Changelog**: https://github.com/Grasscutters/Grasscutter/compare/v1.1.0...v1.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Grasscutters/Grasscutter/releases/tag/v1.2.0'>Grasscutter | v1.2.0</a>.</em>

**Comment by Keyezi** [comment]
你说热爱的

**Comment by Keyezi** [comment]
你所热爱的就是你的生活

**Reply by zhaodice** [reply]
叔叔我啊，真的要生气了

**Reply by HanJQSang** [reply]
怎么国人都跑到Github上来了

**Comment by lunaticwhat** [comment]
ooo i just realized #1234 so cool number

**Comment by chengchuan63** [comment]
这个东西怎么弄？

**Reply by Keyezi** [reply]
先这样，在那样就可以了

**Reply by CCasusensa** [reply]
笑死，有答跟没答一样，但我喜欢

**Reply by CCasusensa** [reply]
就跟平常GC架設一樣，只是切到stable分支 下載而已
https://github.com/Grasscutters/Grasscutter/tree/stable
